newsroom/products/products.py

```diff
@@ -40,7 +40,13 @@ class ProductsResource(newsroom.Resource):
 
 
 class ProductsService(newsroom.Service):
-    pass
+    def on_deleted(self, doc):
+        lookup = {"products._id": doc["_id"]}
+        for resource in ("users", "companies"):
+            items = superdesk.get_resource_service(resource).get(req=None, lookup=lookup)
```

Comment:
Will this be affected by pagination? I don't see an issue with `companies` resource, but maybe the `users` resource could have more than the default page size.

Reply:
there should be no limit if not set on the request